doc/developer/guide-changes.md

```diff
@@ -0,0 +1,634 @@
+reviewer says "+1" or, more explicitly, "please wait for someone else to
+accept", that means the reviewer has issued only a partial approval, and you'll
+need to wait for another reviewer to issue a full approval. Partial approvals
+are most common on large PRs, where you need to gather consensus from several
+different code owners. (This is a great reason to prefer small PRs when
+possible, since you can gather full approvals from various owners in parallel.)
+
+#### Merge skew
+
+Sometimes, two PRs are fine when tested isolation, but the combination of the
+two PRs landing together on `main` can cause tests to fail—or even cause
+Materialize to fail to compile entirely. We call this situation "merge skew."
+
+When this happens, Buildkite will send a Slack message to #engineering
+indicating that tests failed on `main`. If your PR is implicated in the failure,
+please try to jump on fixing the merge skew as soon as possible, or nominate
+someone else to fix it if you are otherwise occupied.
+
+Merge skew usually has the following form. One PR will add a new call to an
+existing function, while another PR adjusts the signature of that function
+definition. Both changes pass tests in isolation. The changes won't conflict
+according to Git's merge algorithm, because they don't touch the same lines. But
+when both changes are present, the code won't compile, because the new call site
+uses the old signature for the function. The analogous situation arises when
+one PR adds a new test while another PR adjusts the behavior tested by that
+new test.
+
+Fixing merge skew is usually as simple as identifying the above situation
+and adjusting the new function call or test case to account for the change
+in the other PR.
+
+While there _are_ tools that can prevent merge skew, like [Bors], they introduce
+quite a bit of latency and flakiness when merging a PR. As long as merge skew is
+fairly rare—right now it seems to happen once every month or two—net
+productivity is still higher without Bors.
+
+#### Branch restrictions
+
+In very rare cases, it may be necessary to merge a PR whose tests are failing.
+This situation typically only occurs with chicken-and-egg CI configuration
+issues, where PR builds are broken until a fix can be merged to `main`.
+
+To perform a "god mode" merge, you can temporarily lift the branch restrictions
+on the `main` branch. Navigate to the [branch protection rules for the `main`
+branch][branch-protection] and uncheck the "Include administrators" box. Do
+*not* delete the rules or uncheck any other boxes, as it is much harder to
+reverse those changes.
+
+Your PR will be mergeable after you uncheck the "Include administrators" box,
+though you will have to click through several warning screens. Once the PR is
+merged, be sure to re-enable the "Include administrators" setting to prevent
+accidents.
+
+## Reviewing changes
+
+The goals of code review are twofold:
```

Comment:
Is there a third purpose: To socialize changes with other engineers?

Reply:
Sure, done!